pkg/backend/httpstate/client/client.go

```diff
@@ -508,16 +508,26 @@ func (pc *Client) ApplyPolicyPack(ctx context.Context, orgName string, policyPac
 }
 
 // DownloadPolicyPack applies a `PolicyPack` to the Pulumi organization.
-func (pc *Client) DownloadPolicyPack(ctx context.Context, locationPath string) ([]byte, error) {
-	fmt.Println("Downloading policy pack from", locationPath)
+func (pc *Client) DownloadPolicyPack(ctx context.Context, url string) ([]byte, error) {
+	fmt.Println("Downloading policy pack from", url)
 
-	pack := []byte{}
-	err := pc.restCall(ctx, "GET", "/api/"+locationPath, nil, nil, &pack)
+	getS3Req, err := http.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
-		return nil, errors.Wrapf(err, "HTTP GET to download policy pack at %q failed", locationPath)
+		return nil, errors.Wrapf(err, "Failed to download compressed PolicyPack")
 	}
 
-	return pack, nil
+	resp, err := http.DefaultClient.Do(getS3Req)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to download compressed PolicyPack")
+	}
+
+	var tarball []byte
```

Comment:
I think you need to defer closing resp.Body?

Reply:
You do. I meant for this to be signposting that this PR is coming, rather than to be reviewed. I don't even know if this code works at all yet because I don't have the service to test against.